src/main/java/journeymap/client/data/WorldData.java

```diff
@@ -285,10 +287,25 @@ public static String getSafeDimensionName(WorldProvider worldProvider)
 
         try
         {
-            return worldProvider.getDimensionName();
+            if (dimNames == null) {
+                dimNames = new TIntObjectHashMap<String>();
+            }
+            if (dimNames.containsKey(worldProvider.dimensionId)) {
+                return dimNames.get(worldProvider.dimensionId);
+            }
+
+            String langKey = String.format("jm.dimension.%1$d.name", worldProvider.dimensionId);
```

Comment:
this needs to be `jm.common.dimension.%1$d.name` and it needs to be added to at least the en_US.lang there is no translation for this key.

Reply:
It's not supposed to exist by default. In absence, the name is retrieved from dimension provider like before.
I've added a caching layer to prevent repetitive translation failures.

Reply:
Then why add a key? 
you will just see `"jm.dimension.overworld.name"` or `"jm.dimension.somemoddimname.name"`? 
Your code explicitly creates a language key, but there is no value for the key.

Reply:
To combat situation, where the same provider is used to create multiple dimensions (FE multiworld, for example), resulting in multiple identical names appearing in various parts of waypoint management UI.
This could be useful for RP purposes as well, if all you care about is the dimensions' names in JM interface.
I will rename the key tomorrow, as requested.